code-studio/src/js/components/progress/stage_progress.jsx

```diff
@@ -1,14 +1,103 @@
+import Radium from 'radium';
 import React from 'react';
 import { STAGE_PROGRESS_TYPE } from './types';
 import { saveAnswersAndNavigate } from '../../levels/saveAnswers';
+import color from '../../color';
+
+var style = {
+  overviewContainer: {
+    display: 'table-cell',
+    verticalAlign: 'middle',
+    paddingRight: '10px'
+  },
+  headerContainer: {
+    padding: '5px 8px',
+    backgroundColor: color.lightest_gray,
+    border: `1px solid ${color.lighter_gray}`,
+    borderRadius: '5px'
+  },
+  dot: {
+    puzzle: {
+      display: 'inline-block',
+      width: '24px',
+      height: '24px',
+      fontSize: '14px',
+      textAlign: 'center',
+      lineHeight: '24px',
+      borderRadius: '24px',
+      borderWidth: '2px',
+      borderStyle: 'solid',
+      borderColor: color.lighter_gray,
+      margin: '0 -1px',
+      transition: 'background-color .2s ease-out, border-color .2s ease-out, color .2s ease-out',
+      ':hover': {
+        textDecoration: 'none',
+        color: color.white,
+        backgroundColor: color.level_current
+      }
+    },
+    unplugged: {
+      width: 'auto',
+      fontSize: '13px',
+      padding: '0 10px'
+    },
+    assessment: {
+      borderColor: color.assessment
+    },
+    small: {
+      width: '7px',
+      height: '7px',
+      borderRadius: '7px',
+      lineHeight: 'inherit',
+      fontSize: 0
+    },
+    overview: {
+      height: '30px',
+      width: '30px',
+      margin: '2px',
+      fontSize: '16px',
+      lineHeight: '32px'
+    }
+  },
+  status: {
+    submitted: {
+      color: color.white,
+      backgroundColor: color.level_submitted
+    },
+    perfect: {
+      color: color.white,
+      backgroundColor: color.level_perfect
+    },
+    passed: {
+      color: color.white,
+      backgroundColor: color.level_passed
+    },
+    attempted: {
+      color: color.charcoal,
+      backgroundColor: color.level_attempted
+    },
+    not_tried: {
+      color: color.charcoal,
+      backgroundColor: color.level_not_tried
+    },
+    review_rejected: {
+      color: color.white,
+      backgroundColor: color.level_review_rejected
+    },
+    review_accepted: {
+      color: color.white,
+      backgroundColor: color.level_review_accepted
+    }
+  }
+};
```

Comment:
Re: CSS, I miss the "Cascading" part of Cascading Style Sheets.  It used to be that we could set one class on a root element and easily apply a totally different set of styles for many elements (e.g. `#teacher-course` for PLC script overview).

I like that components own all their styles and can be included with no extra dependencies, but it seems like we've lost some of the power of CSS.  Outside CSS can still have unintended effects on components.  Maybe Shadow DOM will make this better in the future?